packages/react-compose/src/next/getSlots.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { getNativeElementProps } from '@uifabric/utilities';
+import { GenericDictionary } from './types';
+import { nullRender } from './nullRender';
+
+export const getSlots = (state: GenericDictionary, slotNames: string[] | undefined) => {
+  const slots: GenericDictionary = {
+    root: state.as || nullRender,
+  };
+  const slotProps: GenericDictionary = {
+    root: getNativeElementProps(state.as, state),
+  };
+
+  for (const name of slotNames!) {
+    const slotDefinition = state[name];
+
+    const slot = (slots[name] = slotDefinition.as && slotDefinition.children ? slotDefinition.as : nullRender);
+
+    if (slots[name] !== nullRender) {
+      const nativeElementProps = (slotProps[name] = getNativeElementProps(slot, slotDefinition));
+
+      if (typeof nativeElementProps.children === 'function') {
+        slots[name] = React.Fragment;
+        nativeElementProps.children = nativeElementProps.children(slots[name], nativeElementProps);
```

Comment:
this doesn't seem quite right.
1. we should set ` slots[name] = React.Fragment;` after this line so `slots[name]` is passed with right/original value
2. we should be passing original slotProps instead of nativeElementProps

Reply:
you're right!